### DISCUSSION TITLE
low-level-rustls example plus websockets

### DISCUSSION THREAD

**Discussion by alexander-buerger-met-no** [tokio-rs/axum#2104] [Q&A]
I tried to add a websocket to the low-level-rustls example. Basically, I copied `ws_handler` and the route from the websockets example to the low-level-rustls example. For some reason, the websocket upgrade is never done. Do I need to do anything extra for websocket support?

**Comment by davidpdrsn** [comment]
Good question! I don't know. In hyper's low level API you generally have to enable upgrades (for example with [`Connection::with_upgrades`](https://docs.rs/hyper/0.14.27/hyper/server/conn/struct.Connection.html#method.with_upgrades)) but I don't know how to do it.

Try asking in `#hyper` in the [tokio discord](https://discord.gg/tokio).
